zcash_test_vectors/zip_0244.py

```diff
@@ -23,6 +25,13 @@
 )
 
 
+def ciphertext_offset(tx_version_group_id):
+    if tx_version_group_id == NU7_VERSION_GROUP_ID:
+        # Compact ends at 84, Memo ends at 596 for V6
+        return 84, 596
+    # Compact ends at 52, Memo ends at 564 for V5
+    return 52, 564
+
```

Comment:
Let's have names for the return values and explicit version for each:
```python
def ciphertext_offset(tx_version_group_id):
    if tx_version_group_id == NU5_VERSION_GROUP_ID || ... :
        # Compact ends at 52, Memo ends at 564 for V5
        return {'compact_end': 52, 'memo_end': 564}
    elif tx_version_group_id == NU7_VERSION_GROUP_ID:
        # Compact ends at 84, Memo ends at 596 for V6
        return {'compact_end': 84, 'memo_end': 596}
    else:
        raise ValueError("Unsupported transaction version group id")
```

Reply:
and move down close to usage, this is not important enough to be at the top of the file.

Reply:
Done: Currently we only have NU5 in the if branch. Ideally we should have NU6 there too, but I don't think the test vectors have any NU6 values

Reply:
Would it be nicer to use NU5_TX_VERSION and NU7_TX_VERSION? That is not directly stored in the transaction though, so the function would have to be defined like this:

```python
def ciphertext_offset(tx_version):
    if tx_version == NU5_TX_VERSION | (1 << 31) :
        # Compact ends at 52, Memo ends at 564 for V5
        return {'compact_end': 52, 'memo_end': 564}
    elif tx_version == NU7_TX_VERSION | (1 << 31):
        # Compact ends at 84, Memo ends at 596 for V6
        return {'compact_end': 84, 'memo_end': 596}
    else:
        raise ValueError("Unsupported transaction version group id")
```

and called as follows

```python
ciphertext_offset(tx.version_bytes())
```

Reply:
yes, it is better.

Reply:
Done